cds_migrator_kit/modules/migrator/log.py

```diff
@@ -0,0 +1,137 @@
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+"""Define Books loggers."""
+import json
+import logging
+import os
+
+from pathlib import Path
+from cds_dojson.marc21.fields.books.errors import ManualMigrationRequired, \
+    MissingRequiredField, UnexpectedValue
+
+from cds_migrator_kit.config import MIGRATION_LOG_FILE, MIGRATION_LOGS_PATH
+from cds_migrator_kit.modules.migrator.errors import LossyConversion
+
+
+def set_logging():
+    """Sets additional logging to file for debug."""
+    logger = logging.getLogger('migrator')
+    logger.setLevel(logging.DEBUG)
+    formatter = logging.Formatter('%(asctime)s - %(name)s - '
+                                  '%(message)s - \n '
+                                  '[in %(pathname)s:%(lineno)d]')
+    fh = logging.FileHandler('migrator.log')
+    fh.setFormatter(formatter)
+    fh.setLevel(logging.DEBUG)
+    logger.addHandler(fh)
+    return logger
+
+logger = logging.getLogger('migrator')
+
+
+class JsonLogger(object):
+    """Log migration statistic to file controller."""
+
+    @staticmethod
+    def get_stat_by_recid(recid, stats_json):
+        """Search for existing stats of given recid."""
+        return next(
+            (item for item in stats_json if item['recid'] == recid), None)
+
+
+    def render_stats(self):
+        """Load stats from file as json."""
+        try:
+            with open(MIGRATION_LOG_FILE, "r") as f:
+                all_stats = json.load(f)
+        except IOError as e:
+            open(MIGRATION_LOG_FILE, "w+").close()
+            all_stats = []
+            logger.error(e.message)
+        except ValueError as e:
+            all_stats = []
+            logger.error(e.message)
+        return all_stats
+
+    def create_output_file(self, recid, output):
+        """Create json preview output file."""
+        print('---------------------FILE CREATING')
+        try:
+            filename = "{0}{1}.json".format(MIGRATION_LOGS_PATH, recid)
+            f = Path(filename)
+            f.touch(exist_ok=True)
+            logger.info(f)
+            logger.info(filename)
+            logger.info(os.path.abspath(__file__))
+            logger.info(os.path.relpath(__file__))
+
+            with open(filename, "w+") as f:
+                json.dump(output, f, indent=2)
+        except Exception as e:
+            print e.message
```

Comment:
It will not work in Python 3, it is not a function, what about changing all of them to print()?